pandas/io/parsers.py

```diff
@@ -2195,6 +2257,168 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
         return values
 
 
+class BytesIOWrapper:
+    """
+    Allows the pyarrow engine for read_csv() to read from string buffers
+    """
+
+    def __init__(self, string_buffer: StringIO, encoding: str = "utf-8"):
+        self.string_buffer = string_buffer
+        self.encoding = encoding
+
+    def __getattr__(self, attr: str):
+        return getattr(self.string_buffer, attr)
+
+    def read(self, size: int = -1):
+        content = self.string_buffer.read(size)
+        return content.encode(self.encoding)
+
+
+class ArrowParserWrapper(ParserBase):
+    """
+    Wrapper for the pyarrow engine for read_csv()
+    """
+
+    def __init__(self, src, **kwds):
+        self.kwds = kwds
+        self.src = src
+
+        ParserBase.__init__(self, kwds)
+
+        encoding = kwds.get("encoding") if kwds.get("encoding") is not None else "utf-8"
+
+        self.usecols, self.usecols_dtype = _validate_usecols_arg(kwds["usecols"])
+        na_values = kwds["na_values"]
+        if isinstance(na_values, dict):
+            raise ValueError(
+                "The pyarrow engine doesn't support passing a dict for na_values"
+            )
+        self.na_values = list(
+            _clean_na_values(
+                kwds["na_values"], keep_default_na=kwds["keep_default_na"]
+            )[0]
+        )
+        if isinstance(self.src, TextIOBase):
+            self.src = BytesIOWrapper(self.src, encoding=encoding)
+
+    def read(self):
+        VERSIONS["pyarrow"] = "0.15.0"
```

Comment:
I think we shouldn't edit the `VERSIONS` dict here in place, as that will modify it also for other functionality that requires pyarrow (but not necessarily 0.15). 
Can you edit `import_optional_dependency` to allow to pass an optional "min_version" instead?